### PR TITLE
Add Release signing using AptReleaseSigningService

### PR DESCRIPTION
**Pull request by quba42**
Based on the POC by @mdellweg in https://github.com/pulp/pulp_deb/pull/154.
(I am literally reusing the same commit with minor adjustments).

In addition this PR adds a dedicated AptReleaseSigningService instead of (mis-)using the AsciiArmoredDetachedSigningService from pulpcore.

### CODE REVIEW

File: pulp_deb/app/models/signing_service.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        """
+        with tempfile.TemporaryDirectory() as temp_directory_name:
+            test_release_path = os.path.join(temp_directory_name, "Release")
+            with open(test_release_path, "wb") as test_file:
+                test_data = b"arbitrary data"
+                test_file.write(test_data)
+                test_file.flush()
+                try:
+                    return_value = self.sign(test_release_path)
+                except JSONDecodeError:
```
Comment (mdellweg) [inline pulp_deb/app/models/signing_service.py]: Can we move that to the `sign` function?

File: pulp_deb/app/tasks/publishing.py [hunk 2]
```diff
@@ -218,15 +229,26 @@ def finish(self):
...
         release_metadata = PublishedMetadata.create_from_file(
             publication=self.publication, file=File(open(release_path, "rb")),
         )
         release_metadata.save()
-        # TODO Sign release
+        if self.signing_service:
+            signed = self.signing_service.sign(release_path)
+            for signature_file in signed["signatures"].values():
+                file_name = os.path.basename(signature_file)
+                relative_path = os.path.join(release_dir, file_name)
```
Comment (mdellweg) [inline pulp_deb/app/tasks/publishing.py]: I think, we don't need to specify this `relative_path` as it will be derived from the `File` object.

Comment (quba42) [reply pulp_deb/app/tasks/publishing.py]: Unless the signing service places the signature files in a different folder from the Release file to be signed, which is currently possible.

File: pulp_deb/app/models/signing_service.py (hunk 1 above)
Comment (quba42) [reply pulp_deb/app/models/signing_service.py]: Done.

File: pulp_deb/app/tasks/publishing.py (hunk 2 above)
Comment (mdellweg) [reply pulp_deb/app/tasks/publishing.py]: And could be a requirement if the script needs to run as another user...

Can we make the testing script actually do this then?

Comment (quba42) [reply pulp_deb/app/tasks/publishing.py]: That makes sense, will do.

Comment (quba42) [reply pulp_deb/app/tasks/publishing.py]: Done.

**Pull request merged**